### PR TITLE
Initial commit with some scaffolding. All dev from here on out will f…

### PR DESCRIPTION
**Pull request by bschroeter**
Here is the initial implementation/skeleton for the API.

This repo will follow the formal issue workflow from here on out.

### CODE REVIEW

File: acme/client.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+        Returns
+        -------
+        dict or requests.Response
+            Response from ME.org.
+        """
+        return self._make_request('post', endpoint='logout')
+    
+
+    def get_model_outputs(self, model_output_id):
```
Comment (SeanBryan51) [inline acme/client.py:160]: We can add some type hints here:
```suggestion
    def get_model_outputs(self, model_output_id: str) -> Union[dict, requests.Response]:
```

File: acme/client.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        """
+        self.base_url = base_url
+        self.headers = dict()
+
+        # Automatically login.
+        if email is not None and password is not None:
+            self.login(email, password)
+
+
+    def _make_request(self, method: str, endpoint: str, data: dict=None, headers: dict=None) -> dict:
```
Comment (SeanBryan51) [inline acme/client.py:33]: We can use the [`typing.Union`](https://docs.python.org/3/library/typing.html#typing.Union) type here:
```suggestion
    def _make_request(self, method: str, endpoint: str, data: dict=None, headers: dict=None) -> Union[dict, requests.Response]:
```

File: acme/client.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+            self.headers.update(auth_headers)
+
+        # Unsuccessful login
+        else:
+
+            raise Exception("Login failed")
+
+
+    def get_file_status(self, file_id: str) -> dict:
```
Comment (SeanBryan51) [inline acme/client.py:125]: We can use the [`typing.Union`](https://docs.python.org/3/library/typing.html#typing.Union) type here:
```suggestion
    def get_file_status(self, file_id: str) -> Union[dict, requests.Response]:
```

File: acme/client.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            Response from ME.org.
+        """
+        return self._make_request(
+            method='get',
+            endpoint='modeloutput/{model_output_id}/files',
+            data=dict(model_output_id=model_output_id)
+        )
+
+
+    def start_analysis(self, model_output_id: str) -> dict:
```
Comment (SeanBryan51) [inline acme/client.py:180]: We can use the [`typing.Union`](https://docs.python.org/3/library/typing.html#typing.Union) type here:
```suggestion
    def start_analysis(self, model_output_id: str) -> Union[dict, requests.Response]:
```

File: acme/client.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+        endpoint : str
+            URL slug to add to the base url.
+        data : dict, optional
+            Key/value pairs of data to send., by default None
+        headers : dict, optional
+            Headers to add to the request, by default None
+
+        Returns
+        -------
+        dict or request.Response
```
Comment (SeanBryan51) [inline acme/client.py:49]:
```suggestion
        dict or requests.Response
```

File: acme/client.py [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+            Response from ME.org.
+        """
+        return self._make_request(
+            method='put',
+            endpoint='modeloutput/{model_output_id}/start'
+            data=dict(model_output_id=model_output_id)
+        )
+    
+
+    def get_model_output_status(self, analysis_id: str) -> dict:
```
Comment (SeanBryan51) [inline acme/client.py:200]: We can use the [`typing.Union`](https://docs.python.org/3/library/typing.html#typing.Union) type here:
```suggestion
    def get_model_output_status(self, analysis_id: str) -> Union[dict, requests.Response]:
```

File: acme/client.py [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+            Response from ME.org.
+        """
+        return self._make_request(
+            method='get',
+            endpoint='/modeloutput/{analysis_id}/status',
+            data=dict(analysis_id=analysis_id)
+        )
+
+
+    def list_endpoints(self) -> dict:
```
Comment (SeanBryan51) [inline acme/client.py:220]: We can use the [`typing.Union`](https://docs.python.org/3/library/typing.html#typing.Union) type here:
```suggestion
    def list_endpoints(self) -> Union[dict, requests.Response]:
```

**Review by SeanBryan51 — CHANGES_REQUESTED** [review]
I've added some suggestions for type hints.

Also, will we eventually handle the `requests.Response` object within client rather than returning it explicitly? E.g. `list_endpoints` returning a `list` instead of `dict | requests.Response`.

File: acme/client.py (hunk 1 above)
Comment (bschroeter) [reply acme/client.py:160]: I left the return types as "either or" as we are still working out the specifics. It is unclear at this stage what the return type will need to be, assuming the call fails then the Response object is useful, however, when they are successful then having the dict returned removes an additional step downstream.

Thanks for the type hint catch, implementing now.

**Comment by bschroeter** [comment]
> I've added some suggestions for type hints.
> 
> Also, will we eventually handle the `requests.Response` object within the client rather than returning it explicitly? E.g. `list_endpoints` returning a `list` instead of `dict | requests.Response`.

So list_endpoints is just a helper function at this stage for initial development so I can keep an eye on which endpoints become available on ME.org. The whole returned Response object for this endpoint is quite convoluted so not much use to the user wholesale.

I will aim for this to be simplified later, likely returning the URL and expected parameters.

**Review by SeanBryan51 — APPROVED** [review]
I just noticed there is no import for `Union`, i.e.
```python
from typing import Union
```
Other than that, it looks good.

**Pull request merged**